app/src/main/java/de/danoeh/antennapod/view/viewholder/EpisodeItemViewHolder.java

```diff
@@ -225,9 +228,7 @@ public void bindDummy() {
         }
     }
 
-    private void updateDuration(PlaybackPositionEvent event) {
-        int currentPosition = event.getPosition();
-        int timeDuration = event.getDuration();
+    private void updateDuration(int currentPosition, int timeDuration) {
```

Comment:
Shouldn't it be enough to update the position/duration here, without any checks for the download service?

Just something like this:

```
            getFeedItem().getMedia().setDuration(event.getDuration());
            getFeedItem().getMedia().setPosition(event.getPosition());
```

(please also check whether the feedItem has media before setting the position)